internal/services/appconfiguration/app_configuration_key_resource.go

```diff
@@ -267,6 +325,33 @@ func (k KeyResource) Delete() sdk.ResourceFunc {
 	}
 }
 
+func (k KeyResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			rd := metadata.ResourceDiff
+			keyType := rd.Get("type").(string)
+			if keyType == KeyTypeVault {
+				contentType := rd.Get("content_type").(string)
+				if rd.HasChange("content_type") && contentType != VaultKeyContentType {
+					return fmt.Errorf("vault reference key %q cannot have content type other than %q (found %q)", rd.Get("key").(string), VaultKeyContentType, contentType)
```

Comment:
Would it be less confusing to make this property effectively read-only in this case?

Reply:
I realised I never commited the docs changes :) Did that now.